doc/source/working_with_kiwi/xml_description.rst

```diff
@@ -0,0 +1,985 @@
+connection.
+
+.. _xml-description-product-and-namedCollection-element:
+
+The `product` and `namedCollection` element
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+KIWI supports the inclusion of openSUSE products or of namedCollections
+(*patterns* in SUSE based distributions or *groups* for RedHat based
+distributions). These can be added via the `product` and `namedCollection`
+child elements, which both take the mandatory `name` attribute and the
+optional `arch` attribute.
+
+`product` and `namedCollection` can be utilized to shorten the list of
+packages that need to be added to the image description tremendously. A
+named pattern, specified with the namedCollection element is a
+representation of a predefined list of packages. Specifying a pattern will
+install all packages listed in the named pattern. Support for patterns is
+distribution specific and available in SLES, openSUSE, CentOS, RHEL and
+Fedora. The optional `patternType` attribute on the packages element allows
+you to control the installation of dependent packages. You may assign one
+of the following values to the `patternType` attribute:
+
+- `onlyRequired`: Incorporates only patterns and packages that the
+  specified patterns and packages require. This is a "hard dependency" only
+  resolution.
+
+- `plusRecommended`: Incorporates patterns and packages that are required
+  and recommended by the specified patterns and packages.
+
+
+The `ignore` element
+^^^^^^^^^^^^^^^^^^^^
+
+Packages can be explicitly marked to be ignored for installation inside a
+`packages` collection. This useful to exclude certain packages from being
+installed when using patterns with `patternType="plusRecommended"` as shown
+in the following example:
+
+.. code-block:: xml
+
+   <image schemaversion="7.1" name="Fedora-29">
+     <packages type="image" patternType="plusRecommended">
+       <namedCollection name="network-server"/>
+       <package name="grub2"/>
+       <package name="kernel"/>
+       <ignore name="ejabberd"/>
+       <ignore name="puppet-server"/>
+     </packages>
+   </image>
+
+
+Packages can be marked as ignored during the installation by adding a
+`ignore` child element with the mandatory `name` attribute set to the
+package's name. Optionally one can also specify the architecture via the
+`arch` similarly to :ref:`xml-description-package-element`.
+
+.. warning::
+
+   Adding `ignore` elements as children of a `<packages type="delete">` or
+   a `<packages type="uninstall">` element has no effect! The packages will
+   still get deleted.
+
+
+.. [#f1] `RELAX NG <https://en.wikipedia.org/wiki/RELAX_NG>`_ is a
+         so-called schema language: it describes the structure of a XML
+         document.
+
+.. [#f2] A hybrid persistent filesystem contains a copy-on-write file to
+         keep data persistent over a reboot.
+
+.. FIXME: @schaefi: is this correct?
+.. [#f3] Note that the default primary group is taken from the system on
```

Comment:
The default primary group should be taken from the system being build. As it is defined in `/etc/login.defs` and `useradd` is called within a chroot, thus I expect useradd to set the default according the built image defaults.

So I think is the other way round :wink:

Reply:
I'll fix that.